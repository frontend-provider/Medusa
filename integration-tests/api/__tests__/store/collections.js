const { ProductCollection } = require("@medusajs/medusa")
const path = require("path")
const setupServer = require("../../../helpers/setup-server")
const { useApi } = require("../../../helpers/use-api")
const { initDb, useDb } = require("../../../helpers/use-db")

const productSeeder = require("../../helpers/product-seeder")

jest.setTimeout(30000)
describe("/store/collections", () => {
  let medusaProcess
  let dbConnection

  beforeAll(async () => {
    const cwd = path.resolve(path.join(__dirname, "..", ".."))
    dbConnection = await initDb({ cwd })
    medusaProcess = await setupServer({ cwd })
  })

  afterAll(async () => {
    const db = useDb()
    await db.shutdown()
    medusaProcess.kill()
  })

  describe("/store/collections/:id", () => {
    beforeEach(async () => {
      await productSeeder(dbConnection)
    })

    afterEach(async () => {
      const db = useDb()
      await db.teardown()
    })

    it("gets collection", async () => {
      const api = useApi()

      const response = await api.get("/store/collections/test-collection")

      expect(response.data).toMatchSnapshot({
        collection: {
          id: "test-collection",
          created_at: expect.any(String),
          updated_at: expect.any(String),
        },
      })
    })
  })

  describe("/store/collections", () => {
    beforeEach(async () => {
      await productSeeder(dbConnection)
    })

    afterEach(async () => {
      const db = useDb()
      await db.teardown()
    })

    it("lists collections", async () => {
      const api = useApi()

      const response = await api.get("/store/collections")

      expect(response.data).toMatchSnapshot({
        collections: [
          {
            id: "test-collection2",
            created_at: expect.any(String),
            updated_at: expect.any(String),
          },
          {
            id: "test-collection1",
            created_at: expect.any(String),
            updated_at: expect.any(String),
          },
          {
            id: "test-collection",
            created_at: expect.any(String),
            updated_at: expect.any(String),
          },
        ],
        count: 3,
        limit: 10,
        offset: 0,
      })
    })
  })
})
