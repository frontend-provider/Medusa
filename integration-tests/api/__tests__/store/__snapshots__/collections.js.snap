// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/store/collections /store/collections lists collections 1`] = `
Object {
  "collections": Array [
    Object {
      "created_at": Any<String>,
      "deleted_at": null,
      "handle": "test-collection2",
      "id": "test-collection2",
      "metadata": null,
      "title": "Test collection 2",
      "updated_at": Any<String>,
    },
    Object {
      "created_at": Any<String>,
      "deleted_at": null,
      "handle": "test-collection1",
      "id": "test-collection1",
      "metadata": null,
      "title": "Test collection 1",
      "updated_at": Any<String>,
    },
    Object {
      "created_at": Any<String>,
      "deleted_at": null,
      "handle": "test-collection",
      "id": "test-collection",
      "metadata": null,
      "title": "Test collection",
      "updated_at": Any<String>,
    },
  ],
  "count": 3,
  "limit": 10,
  "offset": 0,
}
`;

exports[`/store/collections /store/collections/:id gets collection 1`] = `
Object {
  "collection": Object {
    "created_at": Any<String>,
    "deleted_at": null,
    "handle": "test-collection",
    "id": "test-collection",
    "metadata": null,
    "title": "Test collection",
    "updated_at": Any<String>,
  },
}
`;
